application/modules/shop/models/DiscountType.php

```diff
@@ -34,7 +34,8 @@ public function rules()
             [['name', 'class', 'checking_class'], 'required'],
             [['active', 'sort_order'], 'integer'],
             [['checking_class'], 'string'],
-            [['name', 'class'], 'string', 'max' => 255]
+            [['name', 'class'], 'string', 'max' => 255],
+            [['custom_view', 'add_view'], 'string', 'max' => 255],
```

Comment:
add_view - ignoring if run validation. And need type $model->add_view = '...' insted of $model->attributes = $attrs
p.s. little fix